### DISCUSSION TITLE
Question - onRecordsListRequest Event Hook filter records

### DISCUSSION THREAD

**Discussion by pietru2004** [pocketbase/pocketbase#6070] [Q&A] [answered]
Scenario:
Users are A B and C
User C blocked User A
User A requests messages from chat
User B requests messages from chat
how to filter returned results via event hooks so that 
user A only get's messages from user A and B but not C
user B  get's messages from user A B and C

Does event onRecordsListRequest support something like
```
if record.user is user C and auth.user is user A then e.excludeRecordFromFinalOutput() else e.includeRecordInFinalOutput()

e.next()
```

Is it possible to do something like that via Event Hook ?

**Comment by ganigeorgiev** [comment, marked as answer]
You can modify the `e.result.items` array but what you are trying to do is wrong because you'll mess up with the pagination.

The more appropriate solution would be to use the List API rule but your collections description is too vague to provide anything concrete.

There is always the option to create a custom route to manually return the specific records you want, but again without anything more concrete I doubt that anyone will be able to assist you. I've asked you several times in the past to try to be more specific with your questions, but you refuse to do so and I'll stop engage to avoid further confusion.

**Reply by pietru2004** [reply]
ok thx for help

**Reply by pietru2004** [reply]
the information I can modify e.result.items when I need to is good enough
